modules/javafx.graphics/src/main/java/com/sun/scenario/animation/StepInterpolator.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright (c) 2023, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package com.sun.scenario.animation;
+
+import javafx.animation.Interpolator;
+import java.util.Objects;
+
+/**
+ * Implementation of a step interpolator as described by
+ * <a href="https://www.w3.org/TR/css-easing-1/#step-easing-algo">CSS Easing Functions Level 1</a>
+ */
+public class StepInterpolator extends Interpolator {
+
+    private final int intervals;
+    private final StepPosition position;
+
+    public StepInterpolator(int intervals, StepPosition position) {
+        if (position == StepPosition.NONE && intervals <= 1) {
+            throw new IllegalArgumentException("intervals must be greater than 1");
+        }
+
+        if (intervals <= 0) {
+            throw new IllegalArgumentException("intervals must be greater than 0");
+        }
+
+        this.position = Objects.requireNonNull(position, "position cannot be null");
+        this.intervals = intervals;
+    }
+
+    @Override
+    protected double curve(double t) {
+        boolean before = t < 0;
+
+        if (before) {
+            t = 0;
+        } else if (t > 1) {
+            t = 1;
+        }
+
+        int step = (int)(t * intervals);
+
+        if (position == StepPosition.START || position == StepPosition.BOTH) {
+            ++step;
+        }
+
+        if (before && (t * intervals % 1 == 0)) {
+            --step;
+        }
+
+        if (t >= 0 && step < 0) {
```

Comment:
Yeah, I figured as much.  I just don't like code that you can't get covered in a unit test, or that contradicts itself.  As you can see, I was confused, is it a bug or a feature?  Perhaps a comment then to indicate that it's intentional.

Reply:
This is covered in `StepInterpolatorTest.testStart`: the test will assert that for t<0, the step interpolator produces the expected value. I also added a comment that this is intended.

Reply:
Thanks for adding the comment, it's fine as is, but I expressed myself unclearly.  What I meant is that you can't get full (branch) coverage on the code in the `if`s -- it can't ever happen in its current form -- I didn't mean to imply there was no test for it at all, sorry.